File: sdk/approute-sdk/javascript/src/resources/steamCurrency.ts
import { BaseResource } from "./base-resource.js";
import type { SteamCurrencyRatesResponse } from "../models/index.js";

export interface SteamCurrencyRatesOptions {
  quotes?: string[];
}

/**
 * Resource for the /steam-currency endpoints.
 */
export class SteamCurrencyResource extends BaseResource {
  /**
   * Get Steam currency exchange rates.
   */
  async rates(
    options?: SteamCurrencyRatesOptions,
  ): Promise<SteamCurrencyRatesResponse> {
    const params: Record<string, unknown> | undefined =
      options?.quotes ? { quotes: options.quotes } : undefined;

    return this.transport.request<SteamCurrencyRatesResponse>(
      "GET",
      "/steam-currency/rates",
      params ? { params } : undefined,
    );
  }
}

File: sdk/approute-sdk/python/approute/resources/steam_currency.py
from __future__ import annotations

from approute.models import SteamCurrencyRatesResponse
from approute.resources._base import BaseResource


class SteamCurrencyResource(BaseResource):

    def rates(self, *, quotes: list[str] | None = None) -> SteamCurrencyRatesResponse:
        """Get Steam currency exchange rates."""
        params = {"quotes": quotes} if quotes else None
        data = self._t.request("GET", "/steam-currency/rates", params=params)
        return SteamCurrencyRatesResponse.model_validate(data)
